pkg/helpers/tokencmd/request_token.go

```diff
@@ -449,3 +464,27 @@ func transportWithSystemRoots(issuer string, clientConfig *restclient.Config) (h
 		return nil, err
 	}
 }
+
+// verifyCertChain uses the system trust bundle in order to perform validation
+// of a certificate chain that needs to have a leaf server cert
+func verifyServerCertChain(dnsName string, chain []*x509.Certificate) ([][]*x509.Certificate, error) {
+	intermediates := x509.NewCertPool()
+	var serverCert *x509.Certificate
+
+	for _, c := range chain {
+		if !c.IsCA {
```

Comment:
i know it is weird but server cert can have `IsCA` field set to `true` . I think it is safe to follow what `tls` package is doing

https://github.com/golang/go/blob/08dd4ad7e30313d089793f19baff1855bbaa004e/src/crypto/tls/handshake_client.go#L805-L818

Reply:
That's for the case of self-signed certs, I'll fix that in the next revision